### DISCUSSION TITLE
showing fable-specific errors in fsautocomplete

### DISCUSSION THREAD

**Discussion by joprice** [fable-compiler/Fable#3849] [Q&A]
I'm using fsautocomplete via ionide from vim, and I recently noticed that a few errors that are fable-specific are not shown as errors by the lsp. One example is

`FABLE: Cannot get type info of generic parameter a. Fable erases generics at runtime, try inlining the functions so generics can be resolved at compile time.`

This also seems to apply to transforms such as JSX and errors from plugins. 

Is there any existing setup that can use fable as the compilation source for fsautocomplete, or perhaps fsautocomplete would need to be extended to support this?

**Comment by MangelMaxime** [comment]
We cannot report error to FSAC from Fable.

You could perhaps create a custom [FSharp.Analyzers.SDK](https://github.com/ionide/FSharp.Analyzers.SDK) to mimic how Fable detect the errors / warning.

The way we do it in VSCode, is by using the [`ProblemMatcher` feature of VSCode](https://github.com/ionide/ionide-vscode-fsharp/blob/4586785a8ecbaf2c9811b373e0cfc22bffb089e8/release/package.json#L1682-L1707) which allows to report error based on the terminal output.

> [!NOTE] 
> Looking at the implementation I think the problem matcher only works for Webpack but it needs to be confirmed.

**Comment by joprice** [comment]
That makes sense. I played around with https://github.com/nvimtools/none-ls.nvim but it doesn't support long-lived processes, so I wrote a prototype custom lsp using Ionide.LanguageServerProtocol that mimics the ProblemMatcher behavior.
